src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,222 @@
+                                                j--;
+                                            }
+                                            break;
+                                        }
+                                    }
+
+                                    // If we didn't find a match, default to the last sequence point
+                                    if (j == sequencePointCount) {
+                                        j--;
+                                    }
+
+                                    while (lines[j] == SequencePoint.HiddenLine && j > 0) {
+                                        j--;
+                                    }
+
+                                    if (lines[j] != SequencePoint.HiddenLine) {
+                                        sourceLine = lines[j];
+                                        sourceColumn = columns[j];
+                                    }
+                                    sourceFile = reader.GetString(reader.GetDocument(documents[j]).Name);
+                                }
+                            }
+                            catch {
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the portable PDB reader for the assembly path
+        /// </summary>
+        /// <param name="assemblyPath">file path of the assembly or null</param>
+        /// <param name="inMemorySymbols">in memory PDB address or zero</param>
+        /// <param name="inMemorySymbolsSize">in memory PDB size</param>
+        /// <param name="reader">returns the reader</param>
+        /// <returns>provider</returns>
+        static IDisposable GetReader(string assemblyPath, IntPtr inMemorySymbols, int inMemorySymbolsSize, out MetadataReader reader)
+        {
+            reader = null;
+            MetadataReaderProvider provider = null;
+
+            if (assemblyPath != null) {
+                try {
+                    uint stamp;
+                    int age;
+                    Guid guid;
+
+                    string pdbName = GetPdbPathFromPeStream(assemblyPath, out age, out guid, out stamp);
+                    if (pdbName != null) {
+                        Stream pdbStream = File.OpenRead(pdbName);
+
+                        // Need to always return provider so that it will be disposed
+                        provider = MetadataReaderProvider.FromPortablePdbStream(pdbStream);
+                        MetadataReader rdr = provider.GetMetadataReader();
+
+                        // Validate that the PDB matches the assembly version
+                        if (age == 1) {
+                            if (IdEquals(rdr.DebugMetadataHeader.Id, guid, stamp)) {
+                                reader = rdr;
+                            }
+                        }
+                    }
+                }
+                catch {
```

Comment:
We definitely need the catch here because if the pdb doesn’t exist or the open fails for some other reason and throws an exception it may lead to recursive calls to this code to dump that exception and overflowing the stack.